sw/device/lib/dif/dif_alert_handler.c

```diff
@@ -4,9 +4,923 @@
 
 #include "sw/device/lib/dif/dif_alert_handler.h"
 
+#include "sw/device/lib/base/bitfield.h"
+
 #include "alert_handler_regs.h"  // Generated.
 
 _Static_assert(ALERT_HANDLER_PARAM_N_CLASSES == 4,
                "Expected four alert classes!");
+_Static_assert(ALERT_HANDLER_PARAM_N_ESC_SEV == 4,
+               "Expected four escalation signals classes!");
+
+dif_alert_handler_result_t dif_alert_handler_init(
+    dif_alert_handler_params_t params, dif_alert_handler_t *handler) {
+  if (handler == NULL) {
+    return kDifAlertHandlerBadArg;
+  }
+
+  // For now, the hardware is hardwired to four signals.
+  if (params.escalation_signal_count != 4) {
```

Comment:
That value can vary with the hardware instantiation; the DIF assumes that that value is 4, which we've already static asserted. I feel like a literal is more clear than the generated constant, since it can't change.

Reply:
This isn't the approach the PLIC takes, which we asked you to match. For instance, the PLIC DIF has a `static_assert` about the value of `RV_PLIC_PARAM_NUMSRC` - and then uses that define elsewhere. When there are top-changes, usually the only place that needs to change is the static assert, as we haven't tripped over any new boundaries - for instance, look at the DIF (and test) changes in #3501 - which are very minimal. If you use literals, it's not minimal, and it's easy to miss updating one, which is why we have the defines at all.

Reply:
I'm not sure that follows. It's not just the literals that need to be updated; there is logic that materially assumes that it's 4; for example, which assumes that only one register is defined to hold certain values, which would change from under the DIF if we just updated the `_Static_assert`.

I can change it, but I don't think it's a good idea.

Reply:
Ah, ok. I think it makes more sense, in that case.